integrations/gitlab/gitlab_integration/ocean.py

```diff
@@ -154,7 +183,60 @@ async def on_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
                 logger.info(
                     f"Finished Processing extras for {projects_processed_in_full_batch}/{len(projects)} projects in batch"
                 )
-                yield enriched_projects
+                yield [
+                    enriched_project.asict() for enriched_project in enriched_projects
+                ]
+
+
+@ocean.on_resync(ObjectKind.PROJECTWITHMEMBERS)
+async def resync_project_with_members(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+
+    for service in get_cached_all_services():
+
+        project_with_members_resource_config: GitlabObjectWithMembersResourceConfig = (
+            typing.cast(GitlabObjectWithMembersResourceConfig, event.resource_config)
+        )
+        if not isinstance(
+            project_with_members_resource_config, GitlabObjectWithMembersResourceConfig
+        ):
+            return
+
+        project_with_members_selector = project_with_members_resource_config.selector
+        include_inherited_members = (
+            project_with_members_selector.include_inherited_members
+        )
+        include_bot_members = project_with_members_selector.include_bot_members
+        include_public_email = project_with_members_selector.include_public_email
+
+        async for projects in service.get_all_projects():
+            projects_batch_iter = iter(projects)
+            projects_processed_in_full_batch = 0
+            while projects_batch := tuple(
+                islice(projects_batch_iter, PROJECT_RESYNC_BATCH_SIZE)
+            ):
+                projects_processed_in_full_batch += len(projects_batch)
+                logger.info(
+                    f"Processing extras for {projects_processed_in_full_batch}/{len(projects)} projects in batch"
+                )
+                tasks = [
+                    service.enrich_project_with_extras(project)
+                    for project in projects_batch
+                ]
+                projects_enriched_with_extras = await asyncio.gather(*tasks)
+                logger.info(
+                    f"Finished Processing extras for {projects_processed_in_full_batch}/{len(projects)} projects in batch"
+                )
+                members_tasks = [
+                    service.enrich_object_with_members(
+                        project,
+                        include_inherited_members,
+                        include_bot_members,
+                        include_public_email,
+                    )
+                    for project in projects_enriched_with_extras
+                ]
+                projects_enriched_with_members = await asyncio.gather(*members_tasks)
```

Comment:
what is making sure we are not experiencing rate limits here?

we get 20 projects, and then spawn 20 tasks and in each project it might have multiple members and then for each member batch you are performing another request.